### PR TITLE
Faster default role syncing during webserver start

### PR DESCRIPTION
**Pull request by jedcunningham**
This makes a handful of bigger queries instead of many queries when
syncing the default Airflow roles. On my machine with 5k DAGs, this led
to a reduction of 1 second in startup time (bonus, makes tests faster
too).

### CODE REVIEW

File: airflow/www/security.py [hunk 1]
```diff
@@ -171,19 +171,25 @@ def __init__(self, appbuilder):
             view.datamodel = CustomSQLAInterface(view.datamodel.obj)
         self.perms = None
 
-    def init_role(self, role_name, perms):
```
Comment (ashb) [inline airflow/www/security.py:174]: We possibly need to keep (but deprecate) this method -- as this is part of a quasi-public API and may be in use by people outside of the Airflow codebase.

File: airflow/www/security.py [hunk 2]
```diff
@@ -467,6 +473,34 @@ def get_all_permissions(self) -> Set[Tuple[str, str]]:
             .all()
         )
 
+    def get_all_non_dag_permissionviews(self) -> Dict[Tuple[str, str], PermissionView]:
```
Comment (ashb) [inline airflow/www/security.py]:
```suggestion
    def _get_all_non_dag_permissionviews(self) -> Dict[Tuple[str, str], PermissionView]:
```

maybe?

File: airflow/www/security.py [hunk 3]
```diff
@@ -467,6 +473,34 @@ def get_all_permissions(self) -> Set[Tuple[str, str]]:
...
+                .join(self.viewmenu_model)
+                .filter(~self.viewmenu_model.name.like(f"{permissions.RESOURCE_DAG_PREFIX}%"))
+                .with_entities(
+                    self.permission_model.name, self.viewmenu_model.name, self.permissionview_model
+                )
+                .all()
+            )
+        }
+
+    def get_all_roles_with_permissions(self) -> Dict[str, Role]:
```
Comment (ashb) [inline airflow/www/security.py]:
```suggestion
    def _get_all_roles_with_permissions(self) -> Dict[str, Role]:
```

too?

**Review by ashb — COMMENTED** [review]
Few small changes, but looks good!

File: airflow/www/security.py (hunk 1 above)
Comment (jedcunningham) [reply airflow/www/security.py:174]: Done.

File: airflow/www/security.py (hunk 2 above)
Comment (jedcunningham) [reply airflow/www/security.py]: Done.

File: airflow/www/security.py (hunk 3 above)
Comment (jedcunningham) [reply airflow/www/security.py]: Done.

File: airflow/www/security.py [hunk 4]
```diff
@@ -171,12 +172,26 @@ def __init__(self, appbuilder):
...
+        :param role_name:
+        :param perms:
+        :return:
+        """
+        warnings.warn(
+            "`init_role` has been deprecated. Please use `sync_default_roles` instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        self.sync_default_roles([{'role': role_name, 'perms': perms}])
```
Comment (ashb) [inline airflow/www/security.py]: Oh wait. the "default" part of the name probably doesn't make sense when called with custom role info. Time to rename it?

Comment (jedcunningham) [reply airflow/www/security.py]: Yeah... `sync_roles` is the natural fit, but it's taken. I'll figure something out.

**Review by ashb — APPROVED** [review]

**Comment by ashb** [comment]
@jedcunningham Could you rebase please -- master had a bug.

**Review by ashb — APPROVED** [review]

**Pull request merged**